### ISSUE TITLE
I tried it but...

### ISSUE THREAD

**Issue by steida** [zeit/now-env#23]
- How to use more different environments? like dev, stage, production?
- How to use secrets in dev? It should be possible.
- How to use it with other software (e.g. Prisma) which works with .env file?

Maybe I just do not understand the purpose of this library.

**Comment by sergiodxa** [comment]
> How to use more different environments? like dev, stage, production?

You can customize the `now.json` file to use using an environment variable (check the README for more info).

Note the idea is that your `now.json` env variables should be the same on any environment, and you should only change the secrets, so in your local machine (dev) you have a `now-secrets.json` with some values, in your staging environment you have another `now-secrets.json` with another set of values and in production you set the secrets with `now secrets add` and let Now set them for you (now-env detects if it's running in Now and does nothing).

> How to use secrets in dev? It should be possible.

If you want to use secrets you added to your ZEIT account using `now secret add` it's not possible, they are stored on the Now servers and you can't access them outside a deployment for security reasons.

That's why you need to create a `now-secrets.json` to define the values for those secrets in your local machine, this will let you develop as if you have access to the secrets and instead use development values.

> How to use it with other software (e.g. Prisma) which works with .env file?

If you want to use something that expects a `.env` file you will need to run somehow run a Node.js script to require `now-env` module and let it set the values on `process.env` (and all of that in the same process). You could may be do it using `node -r now-env server.js`, that will require now-env without changing the code of `server.js`, you will need to find a way to load your software in Node (_note this module is for Node.js apps only, you will need to port it to another platform_).

Basically there is not way for Prisma to know you are using this module if they don't manually add support for it, I would recommend you in that case to just use a `.env` file and stitch to whatever the community of the software you are using is using for env variables.

**Comment by steida** [comment]
Perfect answer. Now I understand it. It's a trade-off. For serious deployment, secrets via Now is the must.  Now I run Now deploy from Node, but it's temporary hacks.

```
execSync('now -e API_SECRET -e PRISMA_ENDPOINT -e PRISMA_SECRET', {
  env: process.env,
  stdio: 'inherit',
});
```

Thank you.
